frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/LogsTab.tsx

```diff
@@ -0,0 +1,245 @@
+import React from 'react';
+import {
+  Button,
+  Tooltip,
+  Toolbar,
+  ToolbarContent,
+  ToolbarGroup,
+  ToolbarItem,
+  ToolbarToggleGroup,
+  Alert,
+  TextListItem,
+  TextListItemVariants,
+  TextContent,
+  Spinner,
+} from '@patternfly/react-core';
+import OutlinedPlayCircleIcon from '@patternfly/react-icons/dist/esm/icons/outlined-play-circle-icon';
+import PauseIcon from '@patternfly/react-icons/dist/esm/icons/pause-icon';
+import PlayIcon from '@patternfly/react-icons/dist/esm/icons/play-icon';
+import EllipsisVIcon from '@patternfly/react-icons/dist/esm/icons/ellipsis-v-icon';
+import DownloadIcon from '@patternfly/react-icons/dist/esm/icons/download-icon';
+import { LogViewer } from '@patternfly/react-log-viewer';
+import { PipelineRunTaskDetails } from '~/concepts/pipelines/content/types';
+import SimpleDropdownSelect from '~/components/SimpleDropdownSelect';
+import useFetchLogs from '~/concepts/k8s/pods/useFetchLogs';
+import { usePipelinesAPI } from '~/concepts/pipelines/context';
+import usePodContainers from '~/concepts/k8s/pods/usePodContainers';
+import { PodContainer } from '~/types';
+
+type LogsTabProps = {
+  task: PipelineRunTaskDetails;
+  isHidden: boolean;
+};
+
+const LogsTab: React.FC<LogsTabProps> = ({ task, isHidden }) => {
+  const { namespace } = usePipelinesAPI();
+  const podName = task.runDetails?.status.podName ?? '';
+  const podContainers = usePodContainers(namespace, podName);
+  const [selectedContainer, setSelectedContainer] = React.useState<PodContainer | null>(
+    podContainers[0],
+  );
```

Comment:
`podContainers` will be empty out of the gate... you may need to do something like this to wait for it:

```
const usePodContainerState = (
  podName: string,
): {
  podContainers: PodContainer[];
  selectedContainer: PodContainer | null;
  setSelectedContainer: (podContainer: PodContainer | null) => void;
} => {
  const { namespace } = usePipelinesAPI();
  const podContainers = usePodContainers(namespace, podName);
  const [selectedContainer, setSelectedContainer] = React.useState<PodContainer | null>(null);

  const firstPodContainer = podContainers[0];
  React.useEffect(() => {
    if (!selectedContainer && firstPodContainer) {
      setSelectedContainer(firstPodContainer);
    }
  }, [firstPodContainer, selectedContainer]);

  return { podContainers, selectedContainer, setSelectedContainer };
};
```